plugins/formal-verification/testData/diagnostics/no_contracts/predicates_access.fir.diag.txt

```diff
@@ -0,0 +1,218 @@
+}
+
+function T_class_global$class_B$get$class_B$member_b(special$get$function$subject: Ref): Int
+  requires acc(T_class_global$class_B(special$get$function$subject), wildcard)
+{
+  (unfolding acc(T_class_global$class_B(special$get$function$subject), wildcard) in
+    special$get$function$subject.class_B$member_b)
+}
+
+function T_class_global$class_C$get$class_A$member_a(special$get$function$subject: Ref): Int
+  requires acc(T_class_global$class_C(special$get$function$subject), wildcard)
+{
+  (unfolding acc(T_class_global$class_C(special$get$function$subject), wildcard) in
+    T_class_global$class_B$get$class_A$member_a(special$get$function$subject))
+}
+
+function T_class_global$class_C$get$class_B$member_b(special$get$function$subject: Ref): Int
+  requires acc(T_class_global$class_C(special$get$function$subject), wildcard)
+{
+  (unfolding acc(T_class_global$class_C(special$get$function$subject), wildcard) in
+    T_class_global$class_B$get$class_B$member_b(special$get$function$subject))
+}
+
+function T_class_global$class_C$get$class_C$member_x(special$get$function$subject: Ref): Ref
+  requires acc(T_class_global$class_C(special$get$function$subject), wildcard)
+{
+  (unfolding acc(T_class_global$class_C(special$get$function$subject), wildcard) in
+    special$get$function$subject.class_C$member_x)
+}
+
+predicate T_class_global$class_A(special$class$predicate$subject: Ref) {
+  acc(special$class$predicate$subject.class_A$member_a, wildcard)
+}
+
+predicate T_class_global$class_B(special$class$predicate$subject: Ref) {
+  acc(special$class$predicate$subject.class_B$member_b, wildcard) &&
+  acc(T_class_global$class_A(special$class$predicate$subject), wildcard)
+}
+
+predicate T_class_global$class_C(special$class$predicate$subject: Ref) {
+  acc(special$class$predicate$subject.class_C$member_x, wildcard) &&
+  acc(T_class_global$class_A(special$class$predicate$subject.class_C$member_x), wildcard) &&
+  acc(T_class_global$class_D(special$class$predicate$subject), wildcard) &&
+  acc(T_class_global$class_B(special$class$predicate$subject), wildcard)
+}
+
+predicate T_class_global$class_D(special$class$predicate$subject: Ref) {
+  true
+}
+
+method class_D$getter_d(this: Ref) returns (ret: Int)
+  requires acc(T_class_global$class_D(this), wildcard)
+  ensures acc(T_class_global$class_D(this), wildcard)
+
+
+method global$fun_accessNested$fun_take$T_class_global$class_C$return$T_Unit(local$c: Ref)
+  returns (ret$0: dom$Unit)
+  requires acc(T_class_global$class_C(local$c), wildcard)
+  ensures acc(T_class_global$class_C(local$c), wildcard)
+{
+  var local0$temp: Int
+  var anonymous$0: Ref
+  inhale dom$Type$isSubtype((dom$TypeOf$typeOf(local$c): dom$Type), dom$Type$global$class_C())
+  unfold acc(T_class_global$class_C(local$c), wildcard)
+  anonymous$0 := local$c.class_C$member_x
+  inhale dom$Type$isSubtype((dom$TypeOf$typeOf(anonymous$0): dom$Type), dom$Type$global$class_A())
```

Comment:
Is this information we would want to include in the predicate?

Reply:
I need to think about it, but seems a good idea!